### PR TITLE
Milestone 2 update on func train_and_predict_model and related

### PR DESCRIPTION
**Pull request by jossiej00**

### CODE REVIEW

**Comment by jossiej00** [comment]
I am not sure about the test cases and the docker file. Could you please check them for me? Thank you very much!

File: tests/pytest/test-train_and_predict_model.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        algorithm = "KNN"
+        mean_acc = np.zeros((k_1 - 1))
+        self.assertEqual(9, len(train_and_predict_model(algorithm, k_1, X_train, X_test, y_train, y_test, mean_acc)),
+                         msg)
+        mean_acc = np.zeros((k_2 - 1))
+        self.assertEqual(8, len(train_and_predict_model(algorithm, k_2, X_train, X_test, y_train, y_test, mean_acc)),
+                                msg)
+        mean_acc = np.zeros((k_3 - 1))
+        self.assertEqual(7, len(train_and_predict_model(algorithm, k_3, X_train, X_test, y_train, y_test, mean_acc)),
+                                 msg)
```
Comment (poddarswakhar) [inline tests/pytest/test-train_and_predict_model.py:44]: this file should go to the tests folder, and after moving delete he pytest folder

File: src/train_and_predict_model.py [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+# ' @return an array of accuracies in terms of different Ks based on a specified algorithm
+# '
+# ' @export
+# '
+# ' @examples
+# ' TrainAndPredictModel('KNN', 81, X_train, y_train, y_test, mean_acc)
+def train_and_predict_model(algorithm, Ks, X_train, X_test, y_train, y_test, mean_acc):
+
     for n in range(1, Ks):
         if algorithm == 'DT':
```
Comment (poddarswakhar) [inline src/train_and_predict_model.py:25]: advancement not required: probably trimming and checking case sensitive less ly
```suggestion
       if algorithm.lower().strip() == "dt":
```

File: src/train_and_predict_model.py [hunk 3]
```diff
@@ -1,28 +1,34 @@
...
-            decTree = DecisionTreeClassifier(criterion="entropy", max_depth = n)
-            decTree.fit(X_train,y_train)
-            yhat=decTree.predict(X_test)
+            decTree = DecisionTreeClassifier(criterion="entropy", max_depth=n)
+            decTree.fit(X_train, y_train)
+            yhat = decTree.predict(X_test)
+            mean_acc[n - 1] = metrics.accuracy_score(y_test, yhat)
         elif algorithm == 'KNN':
```
Comment (poddarswakhar) [inline src/train_and_predict_model.py:30]: advancement not required: probably trimming and checking case sensitive less ly
```suggestion
        elif algorithm.lower().strip() == "knn":
```

**Review by poddarswakhar — CHANGES_REQUESTED** [review]
For the analysis file, the usage of functions is wrong! i.e. train_and_predict_model is returning mean_acc but nothing is being used or analyzed in the ipynb file, lmk for now if you have any questions on discord!

**Pull request merged**